tests/next_tests/integration_tests/feature_tests/ffront_tests/test_decorator.py

```diff
@@ -64,3 +64,97 @@ def testee(a: cases.IField, out: cases.IField):
 
     # with_grid_type returns a new instance, which is frozen but not compiled yet
     assert testee.with_grid_type(cartesian_case.grid_type)._compiled_program is None
+
+
+def _always_raise_callable(*args, **kwargs) -> None:
+    raise AssertionError("This function should never be called.")
+
+
+@pytest.fixture
+def compile_testee(cartesian_case):
+    @gtx.field_operator
+    def testee_op(a: cases.IField, b: cases.IField) -> cases.IField:
+        return a + b
+
+    @gtx.program(backend=cartesian_case.backend)
+    def testee(a: cases.IField, b: cases.IField, out: cases.IField):
+        testee_op(a, b, out=out)
+
+    return testee
+
+
+@pytest.fixture
+def compile_testee_scan(cartesian_case):
+    @gtx.scan_operator(axis=cases.KDim, forward=True, init=0)
+    def testee_op(carry: gtx.int32, inp: gtx.int32) -> gtx.int32:
+        return carry + inp
+
+    @gtx.program(backend=cartesian_case.backend)
+    def testee(a: cases.KField, out: cases.KField):
+        testee_op(a, out=out)
+
+    return testee
+
+
+def test_compile(cartesian_case, compile_testee):
+    if cartesian_case.backend is None:
+        pytest.skip("Embedded compiled program doesn't make sense.")
+
+    assert compile_testee._compiled_program is None
+    compile_testee.compile(offset_provider_type=cartesian_case.offset_provider)
+    assert compile_testee._compiled_program is not None
+
+    args, kwargs = cases.get_default_data(cartesian_case, compile_testee)
+
+    # make sure the backend is never called
+    object.__setattr__(compile_testee, "backend", _always_raise_callable)
```

Comment:
Maybe `unittest.mock.patch()` is a better fit here (and other similar cases below), since it's just a modification for testing.

Reply:
I'll merge it and try that in the follow-up PR.